### PR TITLE
lib/std: fix ByteWriter.read_from method

### PR DESCRIPTION
**Pull request by pierrec**
When reading from a stream which does not have an available method, ByteWriter would not make any progress if its buffer was empty.

### CODE REVIEW

File: test/unit/stdlib/io/bytestream.c3
Comment (lerno) [inline test/unit/stdlib/io/bytestream.c3:1]: Another way to do this with a test reader is to create another module section. So you have first

`module std::io @test;` with the tests, then you declare *another module section*, like `module std::io;` it is now a different section and has different defaults.

File: test/unit/stdlib/io/bytestream.c3 [hunk 1]
```diff
@@ -35,3 +35,43 @@ fn void! bytewriter_buffer()
 	assert(o == "hello");
 	assert(@catchof(s.write("xxxx")));
 }
+
+// TestReader only has the read method to trigger the path
```
Comment (lerno) [inline test/unit/stdlib/io/bytestream.c3:39]: So basically insert `module std::io;` here.

File: test/unit/stdlib/io/bytestream.c3 [hunk 2]
```diff
@@ -1,6 +1,6 @@
-module std::io @test;
```
Comment (lerno) [inline test/unit/stdlib/io/bytestream.c3:1]: So revert to this

File: test/unit/stdlib/io/bytestream.c3 (hunk 1 above)
Comment (pierrec) [reply test/unit/stdlib/io/bytestream.c3:39]: Ah yes, I keep forgetting about this. Too used to single module per file from other languages :).

**Comment by lerno** [comment]
Thank you for this.

**Pull request merged**